helix-core/src/main/java/org/apache/helix/controller/dataproviders/BaseControllerDataProvider.java

```diff
@@ -428,13 +463,32 @@ public synchronized void setIdealStates(List<IdealState> idealStates) {
   }
 
   /**
-   * Returns the LiveInstances for each of the instances that are curretnly up and running
+   * Returns the LiveInstances for each of the instances that are currently up and running
    * @return
    */
   public Map<String, LiveInstance> getLiveInstances() {
```

Comment:
In which places we will need  LiveInstances containing timeouted instance?

Reply:
In short: everywhere else that isn't critical to "timing out" instances. 

The new `getLiveInstances(boolean)` is minimized to only one place: `MessageGenerationPhase`. This approach minimizes any impact to existing code while also making the "timed-out" instances not reachable from the controller; effectively, those instances will have no operations done on them. This decision is made after @jiajunwang 's comment that we shouldn't directly alter the live instance cache; we should only alter the getter. 

Please let me know if this change makes sense. I believe only changing `MessageGenerationPhase` is sufficient.

Reply:
Actually I'm rethinking this. There could be some external view issue. Will update.

Reply:
Now getLiveInstances will default to excluding timed-out ones.